@app/components/src/AvatarUpload.tsx

```diff
@@ -0,0 +1,208 @@
+  //   console.log(info);
+  //   setFileList([...fileList]);
+  // };
+
+  const beforeUpload = (file: any) => {
+    const fileName = file.name.split(".")[0];
+    const fileType = file.name.split(".")[1];
+    file.uid = getUid(fileName) + "." + fileType;
+    const isJpgOrPng = file.type === "image/jpeg" || file.type === "image/png";
+    if (!isJpgOrPng) {
+      message.error("You can only upload JPG or PNG images!");
+      file.status = "error";
+    }
+    const isLt3M = file.size / 1024 / 1024 < 3;
+    if (!isLt3M) {
+      message.error("Image must smaller than 3MB!");
+      file.status = "error";
+    }
+    return isJpgOrPng && isLt3M;
+  };
+
+  const changeUserAvatar = async (avatarUrl: string | null) => {
+    setSuccess(false);
+    setError(null);
+    try {
+      await changeAvatar({
+        variables: {
+          id: user.id,
+          patch: {
+            avatarUrl,
+          },
+        },
+      });
+      setError(null);
+      setSuccess(true);
+    } catch (e) {
+      setError(e);
+    }
+  };
+
+  const customRequest = (option: RcCustomRequestOptions) => {
+    const { onSuccess, onError, file, onProgress } = option;
+    axios
+      .get(`${process.env.ROOT_URL}/api/s3`, {
+        params: {
+          key: file.uid,
+          operation: "put",
+        },
+      })
+      .then(response => {
+        const preSignedUrl = response.data.url;
+        axios
+          .put(preSignedUrl, file, {
+            onUploadProgress: e => {
+              const progress = Math.round((e.loaded / e.total) * 100);
+              onProgress({ percent: progress }, file);
+            },
+          })
+          .then(response => {
+            if (response.config.url) {
+              changeUserAvatar(response.config.url.split("?")[0]);
+              onSuccess(response.config, file);
+            }
+          })
+          .catch(error => {
+            console.log(error);
+            onError(error);
+          });
+      })
+      .catch(error => {
+        console.log(error);
+        onError(error);
+      });
+  };
+
+  const deleteUserAvatarFromBucket = async () => {
+    if (user && user.avatarUrl) {
+      const key = user.avatarUrl.substring(user.avatarUrl.lastIndexOf("/") + 1);
+      await axios
+        .get(`${process.env.ROOT_URL}/api/s3`, {
+          params: {
+            key: `${key}`,
+            operation: "delete",
+          },
+        })
+        .then(() => {
+          // this isn't confirmation that the item was deleted
+          // only confimation that there wasnt an error..
+          changeUserAvatar(null);
+          return true;
+        })
+        .catch(error => {
+          console.log(JSON.stringify(error));
+          return false;
+        });
+    }
+    return true;
+  };
+
+  const onRemove = async () => {
+    if (await deleteUserAvatarFromBucket()) {
+      setFileList(null);
+    }
+  };
+
+  const uploadButton = (
+    <div>
```

Comment:
For accessibility purposes, this should use a `<button>` element. It should probably also have `aria-label` set to something like "Upload avatar". You can use the [Button component in antd](https://ant.design/components/button/).